src/code/z_actor.c

```diff
@@ -414,294 +420,305 @@ void Actor_GetProjectedPos(PlayState* play, Vec3f* worldPos, Vec3f* projectedPos
-    { { 0, 255, 0, 255 }, { 0, 255, 0, 0 } },         { { 0, 255, 0, 255 }, { 0, 255, 0, 0 } },
-    { { 0, 255, 0, 255 }, { 0, 255, 0, 0 } },
+    { { 0, 255, 0, 255 }, { 0, 255, 0, 0 } },         // ACTORCAT_SWITCH
+    { { 0, 255, 0, 255 }, { 0, 255, 0, 0 } },         // ACTORCAT_BG
+    { { 255, 255, 230, 255 }, { 220, 160, 80, 0 } },  // ACTORCAT_PLAYER
+    { { 0, 255, 0, 255 }, { 0, 255, 0, 0 } },         // ACTORCAT_EXPLOSIVES
+    { { 150, 150, 255, 255 }, { 150, 150, 255, 0 } }, // ACTORCAT_NPC
+    { { 255, 255, 0, 255 }, { 200, 155, 0, 0 } },     // ACTORCAT_ENEMY
+    { { 0, 255, 0, 255 }, { 0, 255, 0, 0 } },         // ACTORCAT_PROP
+    { { 0, 255, 0, 255 }, { 0, 255, 0, 0 } },         // ACTORCAT_ITEMACTION
+    { { 0, 255, 0, 255 }, { 0, 255, 0, 0 } },         // ACTORCAT_MISC
+    { { 255, 255, 0, 255 }, { 200, 155, 0, 0 } },     // ACTORCAT_BOSS
+    { { 0, 255, 0, 255 }, { 0, 255, 0, 0 } },         // ACTORCAT_DOOR
+    { { 0, 255, 0, 255 }, { 0, 255, 0, 0 } },         // ACTORCAT_CHEST
+    { { 0, 255, 0, 255 }, { 0, 255, 0, 0 } },         // ACTORCAT_MAX
 };
 
-void func_800B4F78(TargetContext* targetCtx, s32 type, PlayState* play) {
+void Target_InitLockOn(TargetContext* targetCtx, ActorType type, PlayState* play) {
     TatlColor* tatlColorEntry;
     s32 i;
-    TargetContextEntry* targetEntry;
+    LockOnTriangleSet* lockOnEntry;
 
-    Math_Vec3f_Copy(&targetCtx->targetCenterPos, &play->view.eye);
-    targetCtx->unk48 = 0x100;
+    Math_Vec3f_Copy(&targetCtx->lockOnPos, &play->view.eye);
+    targetCtx->lockOnAlpha = 256;
     tatlColorEntry = &sTatlColorList[type];
-    targetCtx->unk44 = 500.0f;
+    targetCtx->lockOnRadius = 500.0f;
 
-    targetEntry = targetCtx->unk50;
-    for (i = 0; i < ARRAY_COUNT(targetCtx->unk50); i++, targetEntry++) {
-        Target_SetPos(targetCtx, i, 0.0f, 0.0f, 0.0f);
+    lockOnEntry = targetCtx->lockOnEntries;
+    for (i = 0; i < ARRAY_COUNT(targetCtx->lockOnEntries); i++, lockOnEntry++) {
+        Target_SetLockOnPos(targetCtx, i, 0.0f, 0.0f, 0.0f);
 
-        targetEntry->color.r = tatlColorEntry->inner.r;
-        targetEntry->color.g = tatlColorEntry->inner.g;
-        targetEntry->color.b = tatlColorEntry->inner.b;
+        lockOnEntry->color.r = tatlColorEntry->inner.r;
+        lockOnEntry->color.g = tatlColorEntry->inner.g;
+        lockOnEntry->color.b = tatlColorEntry->inner.b;
     }
 }
 
-void Target_SetColors(TargetContext* targetCtx, Actor* actor, s32 type, PlayState* play) {
-    targetCtx->unk0.x = actor->focus.pos.x;
-    targetCtx->unk0.y = actor->focus.pos.y + (actor->targetArrowOffset * actor->scale.y);
-    targetCtx->unk0.z = actor->focus.pos.z;
+void Target_SetColors(TargetContext* targetCtx, Actor* actor, ActorType type, PlayState* play) {
+    targetCtx->fairyPos.x = actor->focus.pos.x;
+    targetCtx->fairyPos.y = actor->focus.pos.y + (actor->targetArrowOffset * actor->scale.y);
+    targetCtx->fairyPos.z = actor->focus.pos.z;
```

Comment:
This sets more than colors, it also sets position. Could there be a simpler name than `Target_SetFairyPosAndColors`?

Reply:
What about `Target_SetFairyState`?

Reply:
Yeah, I think that's fine? I've been starting to look at fairy docs and there is something called `fairyState` (name might change eventually tho), plus this has `Target_` to give it context. So it's probably fine